### ISSUE TITLE
map.jumpTo(...) while map.easeTo(...) is in progress throws a TypeError: this._onEaseFrame is not a function

### ISSUE THREAD

**Issue by ToHold** [maplibre/maplibre-gl-js#3328]
**maplibre-gl-js version**: 3.5.2

**browser**: Chrome, Edge

### Steps to Trigger Behavior

 1. Do an easeTo 
 2. Then do a jumpTo during the ease animation
 
### Link to Demonstration

https://codepen.io/ToHold/pen/mdvWwPg?editors=1111

### Expected Behavior

I wanted to set map pitch at 40 when the user zoom upper than level 14, then get it back to 0 if the user unzoom under 14 level. And I also wanted to move camera at a specific zoom and position.

### Actual Behavior

When the camera ease animation goes upper 14 zoom level the jumpTo is call and cause a crash because easeTo is not finished.

**Comment by ToHold** [comment]
So after reading source code, building and testing, the Camera._stop method is called after the TaskQueue.run method.
And so the task.cancelled is not set to false when TaskQueue.run check if it is. So it calls task's callback that we wanted to stop. But as we delete Camera.onEaseFrame in Camera.onEaseFrame._stop, it is undefined when Camera._renderFrameCallback is called and crash the app.

TaskQueue cancelled check in TaskQueue.run :
https://github.com/maplibre/maplibre-gl-js/blob/ba72cb973840ce2482ccfba30c9f922ea937d75c/src/util/task_queue.ts#L48-L52

Camera._cancelRenderFrame call in Camera._stop :
https://github.com/maplibre/maplibre-gl-js/blob/ba72cb973840ce2482ccfba30c9f922ea937d75c/src/ui/camera.ts#L1386-L1390

TaskQueue.remove that sets task.cancelled to true :
https://github.com/maplibre/maplibre-gl-js/blob/ba72cb973840ce2482ccfba30c9f922ea937d75c/src/util/task_queue.ts#L29-L38

Camera.onEaseFrame call :
https://github.com/maplibre/maplibre-gl-js/blob/ba72cb973840ce2482ccfba30c9f922ea937d75c/src/ui/camera.ts#L1429

**Comment by ToHold** [comment]
An easy way to fix, but I'm not sure if it's the best way, is to remove "delete this._onEaseFrame" from code and let the last _renderFrameCallback passed.

**Comment by HarelM** [comment]
This code is complicated, especially with all the callbacks.
Having said that, isn't adding a setTimeout in your code enough to overcome this issue?

**Comment by ToHold** [comment]
Yes I think, but in sight of keeping great perfs on the app I'm working on and avoiding any crash, I kept looking for a great fix.
It's with this idea that I think I've found a great patch. Let me explain :
Camera.isEasing() is returning !!this._easeFrameId, but in fact as we see in the demonstration this._easeFrameId can be set to the next TaskID even if call Camera._stop(...) due to passing callback before Camera._stop(...), so Camera.isEasing() can also be broken. That's why I added a boolean named this._isEasing set to false by default that is set to true in Camera._prepareEase(...), and set to false in Camera._afterEase(...). And then in Camera._renderFrameCallback() we can simply check if the Camera is easing before calling Camera._onEaseFrame(...).

It looks like this :

```
isEasing() {
        return this._isEasing;
}

_prepareEase(eventData: any, noMoveStart: boolean, currently: any = {}) {
        this._isEasing = true;
.
.
.
}

_afterEase(eventData?: any, easeId?: string) {
        this._isEasing = false;
.
.
.
}

_renderFrameCallback = () => {
        if (this._isEasing) {
            const t = Math.min((browser.now() - this._easeStart) / this._easeOptions.duration, 1);
            this._onEaseFrame(this._easeOptions.easing(t));
            if (t < 1) {
                this._easeFrameId = this._requestRenderFrame(this._renderFrameCallback);
            } else {
                this.stop();
            }
        }
};

```

What do you think about it @HarelM ?

I might suggest a pull request for this.

**Comment by ToHold** [comment]
There already is a boolean named Camera._moving set the same way as my this._isEasing, it could better to use it for the check in Camera._renderFrameCallback().

**Comment by HarelM** [comment]
I think I'll be able to better understand this in a PR instead of like this.

**Comment by ToHold** [comment]
I figured out that this crash also happen with NavigationControl, because it also use an easeTo : 
https://codepen.io/ToHold/pen/vYbmdmJ

**Comment by johncarmack1984** [comment]
Fixed in #3332 (v3.6.1), by the reporter. Side note: #4104 was labeled a duplicate of this one but was filed after the fix, so it is probably its own bug. This one can be closed.
